app/assets/javascripts/training/components/modals/transfer_step2.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from 'react';
+import SelectableBox from '../../../components/common/selectable_box.jsx';
+
+// Choose modules to transfer
+const TransferStep2 = ({ transferInfo, setTransferInfo, step, setStep }) => {
+  const [allModules, setAllModules] = useState([]);
+
+  const handleModuleSelection = (moduleName) => {
+    setTransferInfo((prev) => {
+      const modules = prev.modules.includes(moduleName)
+        ? prev.modules.filter(module => module !== moduleName)
+        : [...prev.modules, moduleName];
+      return { ...prev, modules };
+    });
+  };
+
+  useEffect(() => {
+    const extractModulesFromHtml = () => {
+      const categoryElements = document.querySelectorAll('.training__categories > li');
+      let modulesFromHtml = [];
+
+      categoryElements.forEach((li) => {
+        const categoryName = li.querySelector('.training__category__header h1')?.innerText.trim();
+        if (categoryName === transferInfo.sourceCategory) {
+          modulesFromHtml = Array.from(li.querySelectorAll('.training__categories__modules > li')).map((moduleLi) => {
+            const name = moduleLi.querySelector('.action-card-title')?.innerText.trim();
+            const description = moduleLi.querySelector('.action-card-text')?.innerText.trim();
+            return { name, description };
+          });
+        }
+      });
```

Comment:
Won't there always be only one `sourceCategory`? If so, the `forEach` is kinda pointless. Instead, doing something like

```
const selectedCategory = Array.from(categoryElements).find(
  (li) =>
    li.querySelector('.training__category__header h1')?.innerText.trim() ===
    transferInfo.sourceCategory
)
const modulesFromHtml = Array.from(
  selectedCategory.querySelectorAll('.training__categories__modules > li')
).map((moduleLi) => {
  const name = moduleLi.querySelector('.action-card-title')?.innerText.trim()
  const description = moduleLi
    .querySelector('.action-card-text')
    ?.innerText.trim()
  return { name, description }
})
```
might be easier to reason.

Reply:
yes this is easier but now i have changed the entire approach and using #react_root for getting modules instead of parsing them from the html